ui/src/features/stage/create-stage.tsx

```diff
@@ -60,7 +66,7 @@ const stageFormToYAML = (data: z.infer<typeof wizardSchema>, namespace: string)
     spec: {
       requestedFreight: data.requestedFreight,
       ...(data.promotionTemplateSteps && {
-        promotionTemplate: { spec: { steps: yaml.parse(data.promotionTemplateSteps) } }
+        promotionTemplate: { spec: { steps: yaml.parse(promotionTemplateSteps) } }
```

Comment:
Won't an error be thrown if the schema is invalid?

Reply:
Interestingly no. We had total control of schema in wizard before it reaches here. The stringified steps are valid so parsing is going to be valid in all cases. But I can see that the beforehand stringify operation is redundant so We can remove it. I will update the code